test/activesupport5/encoding_test.rb

```diff
@@ -454,15 +460,19 @@ def test_to_json_works_when_as_json_returns_infinite_number
 
   class NaNNumber
     def as_json(options = nil)
-      { "number" => Float::INFINITY }
+      { "number" => Float::NAN }
     end
   end
 
   def test_to_json_works_when_as_json_returns_NaN_number
     assert_equal '{"number":null}', NaNNumber.new.to_json
   end
 
-  protected
+  def test_to_json_works_on_io_objects
+    assert_equal STDOUT.to_s.to_json, STDOUT.to_json
+  end
```

Comment:
This is the one failing test case btw. It's described in the OP of this PR.

Reply:
I think that might be a change in behaviour or maybe the test was not in the older version of the test suite. Anyway, Oj return `null` for IO objects. I'd have to look into the code to determine why. I can do that later this evening.

Reply:
I see. In rails.c:1442 there is no encoder for file types. Rails wants a string.

Reply:
replace that line with this

```
    dump_as_string,	// RUBY_T_FILE     = 0x0b,
```